test/reductions/grid_search/test_grid_search_arguments.py

```diff
@@ -0,0 +1,290 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import logging
+import numpy as np
+import pandas as pd
+import pytest
+from sklearn.linear_model import LogisticRegression, LinearRegression
+
+from fairlearn.metrics import DemographicParity, BoundedGroupLoss
+from fairlearn.reductions import GridSearch
+from fairlearn.reductions.grid_search.simple_quality_metrics import SimpleClassificationQualityMetric, SimpleRegressionQualityMetric  # noqa: E501
+
+# ==============================================================
+
+# The following are functions which convert ndarrays into other datatypes
+# They are used to generate different argument types for calls to
+# GridSearch
+
+
+def identity(X):
+    return X
+
+
+def pandasdf(X):
+    return pd.DataFrame(X)
+
+
+def pandasseries(X):
+    # Will not work if ndarray has more than one dimension
+    return pd.Series(X)
+
+
+def ndarray2d(X):
```

Comment:
See https://docs.scipy.org/doc/numpy/reference/generated/numpy.expand_dims.html

Reply:
Changed.... I suppose it's nicer than reshaping via assignment to the `shape` property.